### PR TITLE
[Docs] Add docs for LoyaltyRule[Api]

### PR DESCRIPTION
**Pull request by Tomanhez**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Screenshot 2021-01-29 at 10 25 56
Screenshot 2021-01-29 at 10 26 19

### CODE REVIEW

File: docs/book/index.rst [hunk 1]
```diff
@@ -130,6 +130,7 @@ Documentation sections of The Book referring to Sylius Plus features are:
     orders/shipments
     orders/returns
     products/multi_source_inventory
+    loyalty_rule/loyalty_rule
```
Comment (GSadee) [inline docs/book/index.rst:133]:
```suggestion
    loyalty/loyalty_rule
```
?

File: docs/book/loyalty_rule/loyalty_rule.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
+.. rst-class:: plus-doc
+
+Loyalty Rule
+==============
+
+Adding new Loyalty Rule Configuration to API
+--------------------------------------------
+
+After created a new loyalty rule configuration you need add new ``LoyaltyRuleActionDataTransformer`` that implements ``Sylius\Plus\Loyalty\Infrastructure\DataTransformer\LoyaltyRuleActionDataTransformerInterface``
```
Comment (GSadee) [inline docs/book/loyalty_rule/loyalty_rule.rst:9]:
```suggestion
After created a new loyalty rule configuration you need to add a new ``LoyaltyRuleActionDataTransformer`` that implements ``Sylius\Plus\Loyalty\Infrastructure\DataTransformer\LoyaltyRuleActionDataTransformerInterface``
```

File: docs/book/loyalty_rule/loyalty_rule.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            ;
+        }
+
+        public function supportsTransformation(string $actionType): bool
+        {
+            return $actionType === 'your_custom_configuration';
+        }
+    }
+
+Next you must to register ``LoyaltyRuleActionYourCustomConfigurationDataTransformer`` as a service with tag `sylius_plus.api.loyalty_rule_action_data_transformer`:
```
Comment (GSadee) [inline docs/book/loyalty_rule/loyalty_rule.rst:45]:
```suggestion
Next you must register ``LoyaltyRuleActionYourCustomConfigurationDataTransformer`` as a service with tag `sylius_plus.api.loyalty_rule_action_data_transformer`:
```

File: docs/book/loyalty/loyalty_rule.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
+.. rst-class:: plus-doc
+
+Loyalty Rule
+==============
+
+Adding new Loyalty Rule Configuration to API
+--------------------------------------------
+
+After created a new loyalty rule configuration you need to add a new ``LoyaltyRuleActionDataTransformer`` that implements ``Sylius\Plus\Loyalty\Infrastructure\DataTransformer\LoyaltyRuleActionDataTransformerInterface``
+To be more flexible, creating new ``LoyaltyRuleConfiguration`` is created by new DTO object: ``Sylius\Plus\Loyalty\Application\DTO\LoyaltyRuleAction`` that has ``configuration`` field as an ``array``
```
Comment (GSadee) [inline docs/book/loyalty/loyalty_rule.rst:10]:
```suggestion
To be more flexible, new ``LoyaltyRuleConfiguration`` is created by new DTO object: ``Sylius\Plus\Loyalty\Application\DTO\LoyaltyRuleAction`` that has ``configuration`` field as an ``array``
```

File: docs/book/loyalty/loyalty_rule.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+Array allow you to add any configuration and in the aftermath you have to transform it object to ``Sylius\Plus\Loyalty\Domain\Model\LoyaltyRuleConfiguration\LoyaltyRuleConfigurationInterface`` instance by your custom  ``LoyaltyRuleActionDataTransformer``
```
Comment (GSadee) [inline docs/book/loyalty/loyalty_rule.rst:11]:
```suggestion
That field allows you to add any configuration and in the aftermath, you have to transform this object to ``Sylius\Plus\Loyalty\Domain\Model\LoyaltyRuleConfiguration\LoyaltyRuleConfigurationInterface`` instance by your custom  ``LoyaltyRuleActionDataTransformer``
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "loyalty-rule-docs"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Review by arti0090 — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Tomasz! :1st_place_medal:

**Pull request merged**